### PR TITLE
executor,bindinfo: set number of rows affected for drop global binding

### LINKED ISSUE #32086: Improve usability of `DROP GLOBAL BINDING...`
## Enhancement

Something like `drop global binding for select 1;` always has `0 rows affected`. So if you mistype the SQL for the bind statement this might not show until you run `show global bindings`.

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

This makes it easier to see if a binding was deleted or not.

```
sql> create global binding for select 1 using select 1 ;
Query OK, 0 rows affected (0.0028 sec)

sql> drop global binding for select 1;
Query OK, 1 row affected (0.0023 sec)   ←------------- One binding removed

sql> drop global binding for select 1;
Query OK, 0 rows affected (0.0027 sec)   ←------------- No bindings removed
```

Issue Number: close #32086

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: bindinfo/handle.go [hunk 1]
```diff
@@ -389,9 +389,11 @@ func (h *BindHandle) DropBindRecord(originalSQL, db string, binding *Binding) (e
 		_, err = exec.ExecuteInternal(context.TODO(), `UPDATE mysql.bind_info SET status = %?, update_time = %? WHERE original_sql = %? AND update_time < %? AND bind_sql = %? and status != %?`,
 			deleted, updateTs, originalSQL, updateTs, binding.BindSQL, deleted)
 	}
+	if err != nil {
+		return 0, err
+	}
 
-	deleteRows = int(h.sctx.Context.GetSessionVars().StmtCtx.AffectedRows())
-	return err
+	return uint64(h.sctx.Context.GetSessionVars().StmtCtx.AffectedRows()), nil
```
Comment (hawkingrei) [inline bindinfo/handle.go:396]: Fix the lint

```
[2022-02-01T17:26:08.972Z] /home/jenkins/agent/workspace/tidb_ghpr_check/go/src/github.com/pingcap/tidb/bindinfo/handle.go:396:15: unnecessary conversion
```

Comment (dveeden) [reply bindinfo/handle.go]: @hawkingrei Done, PTAL

**Review by mjonss — COMMENTED** [review]
LGTM but please add an integration test for the command and the expected affected_rows.

**Review by mjonss — APPROVED** [review]

**Review by CbcWestwolf — APPROVED** [review]

**Pull request merged**